shared/src/lib.rs

```diff
@@ -75,22 +91,87 @@ impl From<SocketAddr> for PravegaNodeUri {
 impl PravegaNodeUri {
     pub fn to_socket_addr(&self) -> SocketAddr {
         // to_socket_addrs will resolve hostname to ip address
-        let mut addrs_vec: Vec<_> = self
-            .0
-            .to_socket_addrs()
-            .expect("Unable to resolve domain")
-            .collect();
-        addrs_vec.pop().expect("get the first SocketAddr")
+        match PravegaNodeUri::uri_parts_from_string(self.to_string()) {
+            Ok(uri_parts) => {
+                let mut addrs_vec: Vec<_> =
+                    format!("{}:{}", uri_parts.authority.unwrap(), uri_parts.port.unwrap())
```

Comment:
Two points:
1. Per: https://uri.thephpleague.com/uri/4.0/components/overview/ athority includes the port. Should this be `host`?
2. Avoid `unwrap` here. We don't want to panic. Ideally the object would be structured such that such cases are impossible at runtime. If we absolutely can't avoid it, then use `.expect`

Reply:
I was being a bit lazy here. Authority also contains userinfo, which I did not want to parse

I could change the name `authority` to `userinfo_host`, to be clearer?

Reply:
regarding the unwrap and panic. The existing `port()` method doesn't check for possible panic due to a non-numeric port number.  I did not want to change the existing semantics on how `port()` works, or the existing expectation that `port` had to be valid to avoid a panic.

Interestingly it appears that `port()` is only used in tests. If we can remove all uses of `port()`, we could  split the URI into just scheme and authority and leave the panic to `SocketAddr`. which would most likely happen at the time of  connection use instead of during config build.

Reply:
PravegaNodeUri has public fn `domain_name()` which is used elsewhere in controller-client and connection_factory.  PravegaNodeUri appears to be used to parse a host name returned from the controller and to associate tls certificates with a host name.

The `domain_name` function returns the value I lazily named `authority`.  The existing `domain_name` function did not parse the uri to extract only the host portion of a URI. That is, it didn't exclude userinfo, which might be provided by an end-user when building a connection (future feature) but probably won't be returned by a controller.

Would it be reasonable to ignore the userinfo issue for now?  

Should field 'authority' be renamed 'domain_name' to match the existing function name, or should function `domain_name` be renamed to `host()` to match the URI specification terminology, or something else?

Reply:
I think we can ignore userinfo. We don't use it, and probably never will.

Reply:
I renamed `authority` to `domain_name` to match other parts of the code base. Its not precise, but it is "same name for the same thing"